hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -384,6 +400,7 @@ private synchronized void flushWrittenBytesToServiceInternal(final long offset,
    * operation FIFO queue.
    */
   private synchronized void shrinkWriteOperationQueue() throws IOException {
+    outputStreamStatistics.queueShrunk();
```

Comment:
Queue is not actually getting shrunk here, rather inside the while loop. I'll change this and the respective tests.